File: Data Structures & Algorithms/combination-target-sum/submission-3.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        nums.sort()
        def dfs(res,cur,index,nums,target):
            if target==0:
                res.append(cur[:])
                return
            for i in range(index,len(nums)):
                if target-nums[i]<0:
                    break
                cur.append(nums[i])
                dfs(res,cur,i,nums,target-nums[i])
                cur.pop()
            return 
        res=[]
        cur=[]
        dfs(res,cur,0,nums,target)
        return res

File: Data Structures & Algorithms/find-duplicate-integer/submission-3.java
class Solution {
    public int findDuplicate(int[] nums) {
        int s=nums[0],f=nums[nums[0]];
        while(s!=f){
            s=nums[s];
            f=nums[nums[f]];
            if(s==f) break;
        }
        s=0;
        while(s!=f){
            s=nums[s];
            f=nums[f];
        }
        return s;
    }
}
